File: webapi/Models/HR/Address.cs
using System.ComponentModel.DataAnnotations;
using System.Diagnostics.CodeAnalysis;
using webapi.Models.System;

namespace webapi.Models.HR
{
    public record Address
    {
        [Key]
        public int Id { get; set; }
        [NotNull]
        public required string StreetAddress { get; set; }
        [NotNull]
        public required string PostalCode { get; set; }
        [NotNull]
        public required string District { get; set; }
        [NotNull]
        public required string Municipilaty { get; set; }
        [NotNull]
        public required string Region { get; set; }
        [NotNull]
        public required string PopulatedPlace { get; set; }
    }


    public record AddressDTO
    {
        public required string StreetAddress { get; set; }
        public required string PostalCode { get; set; }
        public required string District { get; set; }
        public required string Municipilaty { get; set; }
        public required string Region { get; set; }
        public required string PopulatedPlace { get; set; }

    }


    public record AddressUpdateDTO
    {
        public required Int64 UpdateId { get; set; }
        public required AddressDTO Data { get; set; }
    }
}

File: reactapp/src/pages/Employees/models/Address.ts
export interface Address {
    id: int,
    district: string,
    municipilaty: string,
    region: string,
    populatedPlace: string,
    streetAddress: string,
    postalCode: string,
}

export const defaultUpdateAddressDTO: Address = {
    id: 0,
    district: "",
    municipilaty: "",
    region: "",
    populatedPlace: "",
    streetAddress: "",
    postalCode: "",
}

export interface AddressDTO {
    district: string,
    municipilaty: string,
    region: string,
    populatedPlace: string,
    streetAddress: string,
    postalCode: string,
}

export const defaultAddressDTO: AddressDTO = {
    district: "",
    municipilaty: "",
    region: "",
    populatedPlace: "",
    streetAddress: "",
    postalCode: "",
}